docs/BENCHMARK.md

```diff
@@ -0,0 +1,93 @@
+# Addition of `VisitedPcs` trait
```

Comment:
This file should be called something more like `Adding_the_VisitedPcs_Trait.md` as it is not specifically about benchmarking in general, rather this specific effort.